lib/util/s3_setup.js

```diff
@@ -24,3 +24,63 @@ module.exports.detect = function(to, config) {
     }
   }
 };
+
+module.exports.get_s3 = function(config) {
```

Comment:
[praise] this centralization is great - been thinking node-pre-gyp should move to https://github.com/aws/aws-sdk-js-v3 soon to address the complaint of many that the aws-sdk v2 is heavy/leads to large package downloads. So this centralization will help with that future move.